### PR TITLE
FlutterErrorDetails.context docs fix

### LINKED ISSUE #47063: FlutterErrorDetails.context API docs claim it's a string (but it's a DiagnosticsNode)
We should provide a code sample showing what the code should look like to create a FlutterErrorDetails with a context. It used to be just a matter of passing a string but now it's not, even though the API docs talk about a string still.

cc @jacob314

### PR DESCRIPTION
**Pull request by Piinks**
## Description

Fixes a doc regression where FlutterErrorDetails.context is described as a String. Also added some examples as recommended in the bug.

## Related Issues

Fixes #47063

## Tests

N/A

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 1]
```diff
@@ -325,7 +325,33 @@ class ErrorSpacer extends DiagnosticsProperty<void> {
 
 /// Class for information provided to [FlutterExceptionHandler] callbacks.
 ///
-/// See [FlutterError.onError].
+/// {@tool snippet}
+/// This is an example of using [FlutterErrorDetails] when calling
+/// [FlutterError.reportError].
+///
+/// ```dart
+/// _runningAsyncTasks = true;
```
Comment (Piinks) [inline packages/flutter/lib/src/foundation/assertions.dart:333]:
```suggestion
/// final bool _runningAsyncTasks = true;
```

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 2]
```diff
@@ -325,7 +325,33 @@ class ErrorSpacer extends DiagnosticsProperty<void> {
 
 /// Class for information provided to [FlutterExceptionHandler] callbacks.
 ///
-/// See [FlutterError.onError].
+/// {@tool snippet}
+/// This is an example of using [FlutterErrorDetails] when calling
+/// [FlutterError.reportError].
+///
+/// ```dart
+/// final bool _runningAsyncTasks = true;
```
Comment (Piinks) [inline packages/flutter/lib/src/foundation/assertions.dart:333]:
```suggestion
/// bool _runningAsyncTasks = true;
```

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 3]
```diff
@@ -325,7 +325,33 @@ class ErrorSpacer extends DiagnosticsProperty<void> {
 
 /// Class for information provided to [FlutterExceptionHandler] callbacks.
 ///
-/// See [FlutterError.onError].
+/// {@tool snippet}
+/// This is an example of using [FlutterErrorDetails] when calling
+/// [FlutterError.reportError].
+///
+/// ```dart
+/// bool _runningAsyncTasks = true;
```
Comment (chunhtai) [inline packages/flutter/lib/src/foundation/assertions.dart:333]: we can turn this into a real sample with ```{@tool sample --template=...}```

Comment (chunhtai) [reply packages/flutter/lib/src/foundation/assertions.dart:333]: why do we want to include this boolean in this sample?

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 4]
```diff
@@ -382,13 +408,28 @@ class FlutterErrorDetails with Diagnosticable {
...
-  /// The string should be in a form that will make sense in English when
-  /// following the word "thrown", as in "thrown while obtaining the image from
-  /// the network" (for the context "while obtaining the image from the
-  /// network").
+  /// ```dart
+  /// ErrorDescription('while running async test code');
+  /// ```
+  ///
+  /// See also:
+  ///   * [ErrorDescription], which provides an explanation of the problem and its
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/assertions.dart:424]: nit: add a blank line before this one.

File: packages/flutter/lib/src/foundation/assertions.dart (hunk 3 above)
Comment (goderbauer) [reply packages/flutter/lib/src/foundation/assertions.dart:333]: +1 to simplifying this example slightly be removing this variable and also the "return" keywords. Just a try/catch with the error report in the catch block should be enough.

**Review by goderbauer — COMMENTED** [review]
The analyzer is unhappy.

File: packages/flutter/lib/src/foundation/assertions.dart (hunk 3 above)
Comment (Piinks) [reply packages/flutter/lib/src/foundation/assertions.dart]: I simplified the sample. @chunhtai do you mean to make it a sample app that crashes?

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 5]
```diff
@@ -325,7 +325,27 @@ class ErrorSpacer extends DiagnosticsProperty<void> {
...
+///     context: ErrorSummary('while running async test code'),
+///   ));
+/// }
+/// ```
+///
+/// See also:
+///
+///   * [FlutterError.onError], which is called whenever the Flutter framework
+///     catches an error
+
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/assertions.dart:348]: nit: remove blank line

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 6]
```diff
@@ -325,7 +325,27 @@ class ErrorSpacer extends DiagnosticsProperty<void> {
 
 /// Class for information provided to [FlutterExceptionHandler] callbacks.
 ///
-/// See [FlutterError.onError].
+/// This is an example of using [FlutterErrorDetails] when calling
+/// [FlutterError.reportError].
+///
+/// ```dart
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/assertions.dart:331]: Can this be wrapped in a `@{tool ...}` section?

File: packages/flutter/lib/src/foundation/assertions.dart [hunk 7]
```diff
@@ -382,13 +402,29 @@ class FlutterErrorDetails with Diagnosticable {
...
+  /// A [DiagnosticsNode] that provides a human-readable description of where
+  /// the error was caught (as opposed to where it was thrown).
+  ///
+  /// The node, e.g. an [ErrorDescription], should be in a form that will make
+  /// sense in English when following the word "thrown", as in "thrown while
+  /// obtaining the image from the network" (for the context "while obtaining
+  /// the image from the network").
+  ///
+  /// ```dart
+  /// ErrorDescription('while running async test code');
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/assertions.dart:414]: Could this example be slightly more informative by showing how this is used within, e.g. `FlutterError.reportError`?

File: packages/flutter/lib/src/foundation/assertions.dart (hunk 6 above)
Comment (Piinks) [reply packages/flutter/lib/src/foundation/assertions.dart:331]: Sure! Do you mean an app that throws an error and crashes? I can't wrap it in a `{@tool ...}` and pass the analyzer without adding a lot of irrelevant code.

File: packages/flutter/lib/src/foundation/assertions.dart (hunk 7 above)
Comment (Piinks) [reply packages/flutter/lib/src/foundation/assertions.dart:414]: Sure! Should I just use the same sample as above?

File: packages/flutter/lib/src/foundation/assertions.dart (hunk 6 above)
Comment (goderbauer) [reply packages/flutter/lib/src/foundation/assertions.dart:331]: Can you just wrap this in a method to make analyzer happy?

File: packages/flutter/lib/src/foundation/assertions.dart (hunk 7 above)
Comment (goderbauer) [reply packages/flutter/lib/src/foundation/assertions.dart:414]: Similar to that one, maybe one hat actually uses an ErrorDescription.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**